apps/functions-subscription/src/adapters/azure/cosmosdb/activation-request.ts

```diff
@@ -17,12 +17,21 @@ import {
 } from '../../../domain/activation-request';
 import { decodeFromFeed } from './decode';
 import { ActivationJobId } from '../../../domain/activation-job';
+import { cosmosErrorToDomainError } from './errors';
 
 export const makeActivationRequestRepository = (
   db: Database,
 ): ActivationRequestRepository => {
   const container = db.container('activations');
   return {
+    insert: (activationRequest) =>
+      pipe(
+        TE.tryCatch(() => container.items.create(activationRequest), E.toError),
+        TE.flatMapEither(({ resource }) =>
+          pipe(resource, ActivationRequestCodec.decode, E.mapLeft(E.toError)),
+        ),
```

Comment:
**suggestion**: What about using the `decodeFromItem` method and transform the `None` to an error?